pkg/apis/planetscale/v2/vitesskeyspace_methods.go

```diff
@@ -212,15 +208,39 @@ func (c *VitessKeyspaceCondition) StatusDuration() time.Duration {
 }
 
 // DeepCopyConditions deep copies the conditions map for VitessKeyspaceStatus.
-func (s *VitessKeyspaceStatus) DeepCopyConditions() map[VitessKeyspaceConditionType]VitessKeyspaceCondition {
-	if s == nil {
-		return nil
-	}
-	out := make(map[VitessKeyspaceConditionType]VitessKeyspaceCondition, len(s.Conditions))
+func (s *VitessKeyspaceStatus) DeepCopyConditions() []VitessKeyspaceCondition {
+	out := make([]VitessKeyspaceCondition, len(s.Conditions))
 
-	for key, val := range s.Conditions {
-		out[key] = *val.DeepCopy()
+	for _, condition := range s.Conditions {
+		out = append(out, *condition.DeepCopy())
 	}
 
 	return out
 }
+
+// GetCondition provides map style access to retrieve a condition from the conditions list by it's type
+// If the condition doesn't exist, we return nil, false.
+func (s *VitessKeyspaceStatus) GetCondition(ty VitessKeyspaceConditionType) (value *VitessKeyspaceCondition, exists bool) {
```

Comment:
I think it may be bug-prone to have an exported method that returns a pointer into the underlying array. Callers may think it's ok to retain the pointer, but that's not safe in general. Either the caller's value might get unintentionally mutated underneath them, or the caller might intend to mutate the value in the array and fail.

Consider this scenario:

```go
ptr := GetCondition(condType)
SetConditionStatus(otherCondType, ...)

// This may or may not update the value in the array,
// depending on whether SetConditionStatus() above
// allocated a new array while appending to the slice.
ptr.Status = corev1.ConditionTrue
```

I think it'll be useful to have an exported `GetCondition()` method, but it should return a deep copy to be safe. We could convert this version to an unexported method `getCondition()` that's used by both `SetConditionStatus()` and `GetCondition()`.

Reply:
Ahh, I see what you're saying. Ok, will adjust.

Reply:
Adjusted.